### PR TITLE
Add new link utilities, icon link helper, and update colored link helpers

### PR DESCRIPTION
**Pull request by mdo**
Fun update for links in Bootstrap with a few new changes across the project...

### New `.icon-link` helper

This allows you to immediately pair any Bootstrap Icon with a hyperlink and have it spaced, aligned, and styled properly. Includes support for icons before or after the text, and includes a basic hover state. There's opportunity for future hover states here, too. _Might make more sense as a component?_

CleanShot 2022-12-29 at 12 57 29@2x

### Color links are now built with CSS variables

Instead of resetting the `color` property, we reset the `--bs-link-color-rgb` values. I've also added a new `text-decoration-color` property to these helpers that defaults to the current link color in `rgb` form with `--bs-link-underline-opacity` as the alpha layer (defaulting to `1`). This leads me to the next update...

### Introducing several new link utilities

To help modify icon links, colored links, and even regular links, we have new `.link-opacity-*` to change the alpha transparency, `.link-offset-*` to change the distance of the underline, `.link-underline-*` to change the underline color`, and `.link-underline-opacity-*` to change the underline alpha transparency. The link opacity and link underline opacity utilities also have a predefined `:hover` variant as well.

CleanShot 2022-12-29 at 12 57 57@2x

CleanShot 2022-12-29 at 12 58 11@2x

CleanShot 2022-12-29 at 12 58 18@2x

CleanShot 2022-12-29 at 12 58 23@2x

CleanShot 2022-12-29 at 12 58 30@2x

---

As a result, this also removes some other custom `.icon-link` instances from the features snippet example and homepage.

### Live preview

- https://deploy-preview-37762--twbs-bootstrap.netlify.app/docs/5.3/helpers/icon-link/
- https://deploy-preview-37762--twbs-bootstrap.netlify.app/docs/5.3/utilities/link/

### CODE REVIEW

**Comment by mdo** [comment]
Going to ship and iterate on this one :D.

**Comment by patrickhlauke** [comment]
For the icon links:

- suggest doubling up the "Style on hover" to also apply on focus or `:focus-visible`
- i'd add a an info/accessibility callout that pretty much replicates what it says on https://icons.getbootstrap.com/#accessibility - and for the examples you have so far, probably safest to add an `aria-hidden="true"` on the `<svg>`

**Comment by patrickhlauke** [comment]
For the link utilities:

- both for underline color and colored links, worth adding the https://getbootstrap.com/docs/5.3/components/buttons/#conveying-meaning-to-assistive-technologies callout
- for link opacity, worth calling out that obviously this will result in contrast ratios that are well below the WCAG minimum

**Review by patrickhlauke — CHANGES_REQUESTED** [review]
See accessibility-related comments

**Comment by mdo** [comment]
All done, thanks @patrickhlauke! Just pushed an update to address all that.

File: site/content/docs/5.3/utilities/link.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+title: Link
+description: Link utilities are used to stylize your anchors to adjust their color, opacity, underline offset, underline color, and more.
+group: utilities
+toc: true
+added: 5.3
+---
+
+## Link opacity
+
+Change the alpha opacity of the link `rgba()` color value with utilities. Please be aware that changes to a color's opacity can result in contrast ratios below the WCAG minimum.
```
Comment (patrickhlauke) [inline site/content/docs/5.3/utilities/link.md:12]: suggest rewording "in contrast ratios below the WCAG minimum" to something more like

`can lead to links with [*insufficient* contrast]({{< docsref "getting-started/accessibility#color-contrast" >}}).`

File: site/content/docs/5.3/helpers/icon-link.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+toc: true
+added: 5.3
+---
+
+The icon link helper component modifies our default link styles to enhance their appearance and quickly align any pairing of icon and text. Alignment is set via inline flexbox styling and a default `gap` value. We stylize the underline with a custom offset and color. Icons are automatically sized to `1em` to best match their associated text's `font-size`.
+
+Icon links assume [Bootstrap Icons](https://icons.getbootstrap.com) are being used, but you can use any icon or image you like.
+
+{{< callout >}}
+Icons used here are likely to be purely decorative, which means they should be hidden from assistive technologies using `aria-hidden="true"`, as we've done in our examples. For icons that are more than decorative, provide an appropriate text alternative via `alt` or `aria-label` attribute.
```
Comment (patrickhlauke) [inline site/content/docs/5.3/helpers/icon-link.md:16]: the suggestion to use `alt` only works when it's an `` element. for `<svg>`s, you need *both* a `role="img"` and an `aria-label`. Maybe worth adding an example right here, a la `<svg ... role="img" aria-label="..."><use xlink:href="..."></use></svg>`

**Review by patrickhlauke — DISMISSED** [review]

**Comment by mdo** [comment]
Done and done!

**Comment by mdo** [comment]
Going to merge now that feedback is addressed—happy to iterate on exact copy more in follow-ups! Thanks for the review in the mean time <3.

File: scss/helpers/_icon-link.scss [hunk 3]
```diff
@@ -0,0 +1,24 @@
+.icon-link {
+  display: inline-flex;
+  gap: .375rem;
```
Comment (Jehong-Ahn) [inline scss/helpers/_icon-link.scss:3]: Why rem over em?
